### PR TITLE
feat(condo): DOMA-7644 payments tab for marketplace

### PR DESCRIPTION
**Pull request by VKislov**

### CODE REVIEW

File: apps/condo/domains/marketplace/components/MarketplacePaymentsContent.tsx [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+        const allPaymentsSumResponse = client.query({
+            query: SUM_PAYMENTS_QUERY,
+            variables: {
+                where: {
+                    ...searchPaymentsQuery,
+                },
+            },
+            fetchPolicy: 'network-only',
+        })
+        setAllPaymentsSum(get(allPaymentsSumResponse, 'data.sum', 0))
```
Comment (nomerdvadcatpyat) [inline apps/condo/domains/marketplace/components/MarketplacePaymentsContent.tsx:97]: `sum` will be in `data.result.sum`.

I suggest to rewrite it like this:
```suggestion
       client.query({
            query: SUM_PAYMENTS_QUERY,
            variables: {
                where: {
                    organization : { id: organization.id },
                    ...searchPaymentsQuery,
                },
            },
            fetchPolicy: 'network-only',
        })
            .then(allPaymentsSumResponse => {
                setAllPaymentsSum(get(allPaymentsSumResponse, 'data.result.sum', 0))
            })
```

File: apps/condo/domains/marketplace/hooks/useFilters.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import { MARKETPLACE_PAGE_TYPES, MarketplacePageTypes } from '@condo/domains/marketplace/utils/clientSchema'
+
+
+const addressStringContainsFilter = getStringContainsFilter(['invoice', 'property', 'address'])
+const invoiceNumberFilter = getStringContainsFilter(['invoice', 'number'])
+const ticketNumberFilter = getStringContainsFilter(['invoice', 'ticket', 'number'])
+const statusFilter = getStringContainsFilter('status')
+const paymentDateRangeFilter = getDayRangeFilter('date')
+
+export function useFilters (marketplacePageType: MarketplacePageTypes): Array<FiltersMeta<PaymentWhereInput>>  {
```
Comment (nomerdvadcatpyat) [inline apps/condo/domains/marketplace/hooks/useFilters.tsx:30]: We need 3 different useFilters/useTableColumns for 3 tabs (invoices/payments/marketItems).
Use it for payments only pls (remove switch/case)

File: apps/condo/domains/marketplace/hooks/useTableColumns.tsx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    , [search])
+
+    const transactionNumberRender = useCallback(payment => {
+        return getTableCellRenderer({ search })('TODO(DOMA-7495) implement rendering of this field after task completion')
+    }
+    , [search])
+
+
+    return useMemo(() => {
+        switch (marketplacePageType) {
```
Comment (nomerdvadcatpyat) [inline apps/condo/domains/marketplace/hooks/useTableColumns.tsx:64]: same as useFilters

File: apps/condo/domains/marketplace/components/MarketplacePageContent/index.tsx [hunk 4]
```diff
@@ -51,8 +57,37 @@ export const MarketplacePageContent = () => {
...
+    const searchPaymentsQuery = useMemo(() => {
+        return {
+            invoice: {
+                context: {
+                    organization: { id: orgId },
+                },
+            },
+        }},
+    [filters, filtersToWhere, orgId])
+    const sortBy = useMemo(() => sortersToSortBy(sorters) as SortPaymentsBy[], [sorters, sortersToSortBy])
```
Comment (nomerdvadcatpyat) [inline apps/condo/domains/marketplace/components/MarketplacePageContent/index.tsx:90]: This all related to `MarketplacePaymentsContent`. Put it inside the `MarketplacePaymentsContent` component pls

**Review by nomerdvadcatpyat — APPROVED** [review]

**Review by AleX83Xpert — APPROVED** [review]

**Pull request merged**